### ISSUE TITLE
Desktop client automatically fast-forwarding current branch

### ISSUE THREAD

**Issue by jayrhynas** [desktop/desktop#8933]
### Describe the bug

The GitHub desktop app seems to be automatically fetching and fast-forwarding the current branch, even if there are local changes. This results in git reporting many "diffs" that would undo the remote changes, since the unchanged working directory doesn't match the branch tip.

This seems to be a regression of #8155.

### Version & OS

Version 2.2.4
macOS 10.15.2

### Steps to reproduce the behavior

1. Clone a repo on machine A from some remote
2. Push changes to the remote from machine B
3. Focus the GitHub desktop client and wait for the auto-fetch to run.

### Expected behavior

Changes are fetched in the background but the current branch isn't affected.

### Actual behavior

Changes are fetched and the current branch is fast-forwarded (if possible). This results in git reporting many "diffs" that would undo the remote changes, since the unchanged working directory doesn't match the branch tip.

**Comment by billygriffin** [comment]
@jayrhynas Thanks for the issue, and I'm sorry you're experiencing this. Does it seem to manifest in the same way as the reports in #8155, or is there anything different in your case from those reports? 

Would you mind including a log file (`Help` -> `Show Logs in Finder`) from a day this occurred for you?

**Comment by jayrhynas** [comment]
It seems to be the same as #8155.
I've attached the logs from today (I just redacted the username since this is from my co-worker's machine)
[2020-01-16.desktop.production.log](https://github.com/desktop/desktop/files/4073247/2020-01-16.desktop.production.log)

**Comment by liran-co** [comment]
+1 Having this issue as well.

**Comment by steveward** [comment]
@liran-co would you be able to upload your log file from GitHub Desktop as well (from the day this issue occurred)? I'd like to take a look to see if there is anything helpful there.

@jayrhynas and @liran-co I would also be interested if you are able to reliably reproduce this in a test repository (if you have the time to). I've made attempts to try and trigger this edge case but so far have been unsuccessful. There may be a detail I'm missing in the reproduction steps.

**Comment by jayrhynas** [comment]
I haven't been able to reproduce it intentionally but it happens multiple times a week to multiple co-workers.

**Comment by liran-co** [comment]
Log file below. I think it's safe to ignore all the authentication errors in there (not sure why they are in the logs...I'm authenticated just fine). For me this is happening nearly every time a branch is updated.

[2020-03-06.desktop.production.log](https://github.com/desktop/desktop/files/4301005/2020-03-06.desktop.production.log)

**Comment by tierninho** [comment]
In the logs it says:
```
Host key verification failed.
fatal: Could not read from remote repository.
```
which usually indicates an issue with SSH -- did you clone this repository in GitHub Desktop, or did you clone it from another client using SSH? You can check the remote URL the repository is using from GitHub Desktop by going to the file menu and selecting `Repository > Repository Settings > Remote`. 

There are two solutions:

1. Switch the remote URL to use HTTPS instead of SSH.

2. Get SSH working. The `host key verification failed` error means that the host key isn't in your `known_hosts` file, with `ssh-keyscan [enter_host] >> ~/.ssh/known_hosts`.

Hope that helps. Let us know if that doesn't get things working.

**Comment by liran-co** [comment]
Yea, that did it (had an old repo in there with SSH). In any case, this happened again today around 2020-03-07T15:52. Log file attached.

[2020-03-07.desktop.production.log](https://github.com/desktop/desktop/files/4302087/2020-03-07.desktop.production.log)

**Comment by tierninho** [comment]
@liran-co Please confirm if it is happening with the same repo or another repo. Also, which solution did you use to initially resolve it? Thanks.

**Comment by liran-co** [comment]
I switched the remote to HTTPS to solve those authentication issues. That wasn't the repo with the issue though.

**Comment by tierninho** [comment]
Did HTTPS not resolve the issue when it happened again? Thanks in advance for the clarity.

**Comment by liran-co** [comment]
No, it did not. As a possibility helpful note: I did notice it happens pretty consistently when switching repos in the desktop app.

**Comment by steveward** [comment]
@jayrhynas and @liran-co could you share how many branches you have in the repository where you are seeing this issue occur? We think the issue may be related to having 20+ branches in a repository.

**Comment by liran-co** [comment]
28 branches :/

**Comment by jayrhynas** [comment]
Also ~100 branches for me

**Comment by jayrhynas** [comment]
> Also ~100 branches for me

Would the issue be resolved by removing the local branches but keeping the remote ones? That would be a quicker solution for us.

**Comment by niik** [comment]
Given that we haven't had any reports of this in a few years I'm going to assume that the fast-forwarding rework in https://github.com/desktop/desktop/pull/11387 and https://github.com/desktop/desktop/pull/11463 took care of this. If you're still encountering this please open a new issue.
